integration/utils/contLoadGenerator.go

```diff
@@ -411,3 +403,186 @@ func (lg *ContLoadGenerator) sendTx(workerID int) {
 		wg.Wait()
 	}
 }
+
+func (lg *ContLoadGenerator) sendAddKeyTx(workerID int) {
+	// TODO move this as a configurable parameter
+	numberOfKeysToAdd := 40
+	blockRef, err := lg.blockRef.Get()
+	if err != nil {
+		lg.log.Error().Err(err).Msgf("error getting reference block")
+		return
+	}
+
+	lg.log.Trace().Msgf("getting next available account")
+
+	acc := <-lg.availableAccounts
+	defer func() { lg.availableAccounts <- acc }()
+
+	lg.log.Trace().Msgf("creating add proposer key script")
+	keys := make([]*flowsdk.AccountKey, 0)
+	for i := 0; i < numberOfKeysToAdd; i++ {
+		keys = append(keys, lg.serviceAccount.accountKey)
+	}
+
+	addKeysTx, err := lg.scriptCreator.AddKeysToAccountTransaction(*(acc.address), keys)
+	if err != nil {
+		lg.log.Error().Err(err).Msgf("error creating token transferscript")
+		return
+	}
+
+	lg.log.Trace().Msgf("creating transaction")
+
+	addKeysTx.SetReferenceBlockID(blockRef).
+		SetProposalKey(*acc.address, 0, acc.seqNumber).
+		SetPayer(*acc.address).
+		AddAuthorizer(*acc.address)
+
+	lg.log.Trace().Msgf("signing transaction")
+	acc.signerLock.Lock()
```

Comment:
Is the underlying crypto implementation already safe for concurrent use? Maybe a question for @tarakby

Reply:
Either way, this signing code is duplicated in a few places so it might be better to encapsulate this as a method on `flowAccount`, then can just do:
```go
err := acc.SignTransaction(addKeysTx)
```

Reply:
The account signatures using ECDSA should be thread safe yes.

Reply:
I changed it for most of the places except the createAccounts method, I want to rebuild that part soon.